octodns/provider/base.py

```diff
@@ -57,19 +52,44 @@ def _process_desired_zone(self, desired):
         for record in desired.records:
             if record._type == 'PTR' and len(record.values) > 1:
                 # replace with a single-value copy
-                self.log.warn('does not support multi-value PTR records; '
-                              'will use only %s for %s', record.value,
-                              record.fqdn)
+                msg = 'multi-value PTR records not supported for {}' \
+                    .format(record.fqdn)
+                fallback = 'falling back to single value, {}' \
+                    .format(record.value)
+                self.supports_warn_or_except(msg, fallback)
                 record = record.copy()
                 record.values = [record.value]
 
             new_desired.add_record(record)
 
         return new_desired
 
+    def _include_change(self, change):
+        '''
+        An opportunity for providers to filter out false positives due to
+        peculiarities in their implementation. E.g. minimum TTLs.
+        '''
+        return True
+
+    def _extra_changes(self, existing, desired, changes):
+        '''
+        An opportunity for providers to add extra changes to the plan that are
+        necessary to update ancillary record data or configure the zone. E.g.
+        base NS records.
+        '''
+        return []
+
+    def supports_warn_or_except(self, msg, fallback):
```

Comment:
I've split the message param into two parts. The first always applies for both exceptions and log warnings as it's the "problem". The `fallback` bit only gets used in log warnings when we've worked around the lack of support. I think that will make things less confusing.

Reply:
This is exactly what I was thinking, but thought it might be an unnecessary complexity. But looks great now 👍